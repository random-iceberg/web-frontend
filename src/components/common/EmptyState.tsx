/*
Draft Doc

This Component is for Empty States.
*/
import React from "react";

interface EmptyStateProps {
  message: string;
}

const EmptyState: React.FC<EmptyStateProps> = ({ message }) => {
  return <div className="p-8 text-center bg-gray-50 rounded-lg">{message}</div>;
};

export default EmptyState;
