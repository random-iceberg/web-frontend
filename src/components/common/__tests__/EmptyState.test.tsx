import React from "react";
import { render } from "@testing-library/react";
import EmptyState from "../EmptyState";

describe("EmptyState Component", () => {
  it("renders correctly with a message", () => {
    const { asFragment } = render(<EmptyState message="No data available" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correctly with a different message", () => {
    const { asFragment } = render(
      <EmptyState message="Search results not found" />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
