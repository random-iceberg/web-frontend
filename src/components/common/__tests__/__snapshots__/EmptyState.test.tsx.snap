// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Component renders correctly with a different message 1`] = `
<DocumentFragment>
  <div
    class="p-8 text-center bg-gray-50 rounded-lg"
  >
    Search results not found
  </div>
</DocumentFragment>
`;

exports[`EmptyState Component renders correctly with a message 1`] = `
<DocumentFragment>
  <div
    class="p-8 text-center bg-gray-50 rounded-lg"
  >
    No data available
  </div>
</DocumentFragment>
`;
